File: TimeConversion.java
public class TimeConversion {
    static String timeConversion(String s) {
        String[] str = s.split(":")
        int hour = Integer.parseInt(str[0]);

        String min = str[1];
        String secPeriod = str[2];
        String sec = str[2].substring(0, secPeriod.length() - 2);

        String newTimeINString = "";

        if ((0 <= hour && hour < 12) && (period.equalsIgnoreCase("AM"))) {
            newTimeINString = String.format("%02d", hour) + ":" + min + ":" + sec;
        } else if ((0 <= hour && hour < 12) && period.equalsIgnoreCase("PM")) {
            newTimeINString = (12 + hour) + ":" + min + ":" + sec;
    
        } else if ((hour == 12) && (period.equalsIgnoreCase("AM"))) {
            newTimeINString = "00" + ":" + min + ":" + sec;
        } else if ((hour == 12) && (period.equalsIgnoreCase("PM"))) {
            newTimeINString = hour + ":" + min + ":" + sec;
        }
        return newTimeINString;
        }
    }
}

File: timeconversion.py
def timeConversion(s):
    AM_PM = s[-2:]                                            #[-2:] means the last two numbers
    s = s[:8]
    hh, mm, ss = [int(x) for x in s.split(":")]               #splitting the array and letting the vals be called hh, mm & ss.
                                        
    if AM_PM == 'PM' and hh != 12:                          
        return('{:02}:{:02}:{:02}'.format(hh+12, mm, ss))     #if its pm and hour not set to 12 we set it. 
    elif AM_PM == 'AM' and hh == 12:                          #if its am and hour not set to 00 we set it.
        return('{:02}:{:02}:{:02}'.format(0, mm, ss))         
    else:                                                         
        return('{:02}:{:02}:{:02}'.format(hh, mm, ss))        #finally we return the list
